jscomp/ml/variant_coercion.ml

```diff
@@ -0,0 +1,38 @@
+
+type variant_representation = String | Int | Float | Other
+
+let find_as_attribute_payload (attributes : Parsetree.attribute list) =
```

Comment:
This logic should be taken from somewhere else: shared with where "as" is already handled in the compiler. We can't have duplication or the complexity will likely lead to bugs on future updates.
This might require a little refactoring.

Reply:
I found no existing general fn for extracting the as-attribute, but I did find other functions for the payload itself which I now use. Looking good?

Reply:
Looks good!